tls/s2n_ktls.h

```diff
@@ -17,19 +17,32 @@
 
 #include "tls/s2n_config.h"
 
+/* Define headers needed to enable and use kTLS.
+ *
+ * The inline header definitions are required to compile kTLS specific code.
+ * kTLS has been tested on linux. For all other platforms, kTLS is marked as
+ * unsupported, and will return an unsupported error.
+ */
+#if defined(__linux__)
+    #define S2N_KTLS_SUPPORTED true
+    #include "tls/s2n_ktls_linux.h"
+#else
+    #define S2N_KTLS_SUPPORTED false
+    #include "tls/s2n_ktls_unsupported.h"
+#endif
```

Comment:
Do you expect these headers to get significantly larger? Right now I think you're adding complexity with the separate headers, and it'd be clearer to just if/else here for the values that vary.

Reply:
https://github.com/toidiu/s2n-tls/blob/ae93e16b2d67f3e68a2dc7d7deaffc09fa1509d4/tls/s2n_ktls_linux.h

Cameron recommended this would be a nice way to represent it. I choose to define all the values here and also 0-init the ones for unsupported platforms (more likely to fail).

Reply:
Minimally, move any value that's the same in both headers out of the separate headers. They should contain only values that differ between supported vs unsupported. That'll make it clear what's different.

So ignoring all the common values, I'm torn on whether I prefer separate headers or same header. It's borderline too small for separate headers. I'd probably lean towards one header.

Especially at this stage, where you only have two values that differ. Maybe do one header for now and revisit whether you need separate headers in a later PR as you add more values.